src/poprox_storage/repositories/pools.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from uuid import UUID, uuid4
+
+from sqlalchemy import (
+    Connection,
+    Table,
+    desc,
+    select,
+)
+from sqlalchemy.dialects.postgresql import insert
+
+from poprox_concepts.domain import Article
+from poprox_storage.repositories.articles import _fetch_articles
+from poprox_storage.repositories.data_stores.db import DatabaseRepository
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+class DbCandidatePoolRepository(DatabaseRepository):
+    def __init__(self, connection: Connection):
+        super().__init__(connection)
+        self.tables: dict[str, Table] = self._load_tables("articles", "candidate_pools", "candidate_articles")
+
+    def store_candidate_pool(self, pool_type: str, articles: list[Article]):
+        candidate_pools_table = self.tables["candidate_pools"]
+        candidate_articles_table = self.tables["candidate_articles"]
+
+        candidate_pool_id: UUID = uuid4()
+
+        set_insert_stmt = insert(candidate_pools_table).values(
+            {"candidate_pool_id": candidate_pool_id, "pool_type": pool_type}
+        )
+        self.conn.execute(set_insert_stmt)
+
+        insert_stmt = (
+            insert(candidate_articles_table)
+            .values(
+                [{"candidate_pool_id": candidate_pool_id, "article_id": article.article_id} for article in articles]
+            )
+            .on_conflict_do_nothing(constraint="uq_candidate_articles")
```

Comment:
I don't think `on_conflict_do_nothing` should do anything here outside of error cases:

- two articles passed in. Shouldn't happen, this is as good of a plan as any.
- Duplicate uuid4. Not worth adjusting plans for.

Probably fine to keep it, or make it an error because if it happens something screwy is going on.

Reply:
I think the impact is just that passing the same article in twice will silently ignore the duplication. I can't see how that would happen and I'm not sure whether it's better to raise an error (or not.)